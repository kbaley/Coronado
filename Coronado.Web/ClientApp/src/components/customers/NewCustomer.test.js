import React from 'react';
import ConnectedNewCustomer, { NewCustomer } from './NewCustomer';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();

describe('NewCustomer tests', () => {
  it('should render properly', () => {
    const tree = renderer
      .create(<NewCustomer />)
    expect(tree.toJSON()).toMatchSnapshot();
  }); 

  it('should map customers to props', () => {
    const customers = [{customerId: "moo", name: "cust1"}];
    const store = mockStore({customers});
    
    const wrapper = shallow(<ConnectedNewCustomer store={store} />);

    expect(wrapper.props().customers).toHaveLength(1);
    expect(wrapper.props().customers).toBe(customers);
  });
});