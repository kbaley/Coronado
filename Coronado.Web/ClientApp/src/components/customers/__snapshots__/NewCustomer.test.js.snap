// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCustomer tests should render properly 1`] = `
<span>
  <span
    className="new-icon icon new-customer glyphicon glyphicon-plus-sign"
    onClick={[Function]}
  />
</span>
`;
